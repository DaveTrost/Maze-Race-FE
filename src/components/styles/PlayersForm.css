@font-face {
  font-family: "Maze"; 
  src: url('../../assets/fonts/Maze.ttf')
}

.PlayersForm {
  display:flex;
  flex-direction: column;
  flex-wrap: wrap;
  justify-content: center;
  align-items: center;
  margin: 0;
}

.PlayerNumber {
  width: 10vw;
  height: 6vh;
  margin: 1vh;
  font-family: "KARNIVOT";
  font-size: 5vh;
  text-align: center;
  color:#00c3ff;
  background: rgb(0, 0, 0);
  outline-color: #ff00ee;
  border: 1px solid white;
  border-radius: 2%;
}