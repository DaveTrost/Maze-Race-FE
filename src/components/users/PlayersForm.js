import React, { useState } from 'react';
import PropTypes from 'prop-types';
import styles from '../styles/PlayerForm.css';

const PlayersForm = ({ handleSubmit, type }) => {

  const [players, setPlayers] = useState('');

  return (
    <form className={styles.PlayersForm} onSubmit={(event) => handleSubmit(event, players)}>
      <input type="number" name="quantity" min="1" max="8" value={players} onChange={({ target }) => setPlayers(target.value)} />
      <br></br>
      <button>Start</button>
    </form>
  );

};

PlayersForm.propTypes = {
  handleSubmit: PropTypes.func.isRequired,
  type: PropTypes.string.isRequired
};

export default PlayersForm;
