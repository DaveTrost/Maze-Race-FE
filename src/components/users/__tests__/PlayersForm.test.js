import React from 'react';
import { shallow } from 'enzyme';
import PlayersForm from '../PlayersForm.js';

describe('PlayersForm component', () => {
  it('renders PlayersForm', () => {
    const wrapper = shallow(<PlayersForm />);
    expect(wrapper).toMatchSnapshot();
  });
});
