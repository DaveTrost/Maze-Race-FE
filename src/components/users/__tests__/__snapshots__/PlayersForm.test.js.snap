// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayersForm component renders PlayersForm 1`] = `
<form
  className="PlayersForm"
  onSubmit={[Function]}
>
  <input
    className="PlayerNumber"
    max="8"
    min="1"
    name="quantity"
    onChange={[Function]}
    type="number"
    value=""
  />
  <br />
  <button>
    Start
  </button>
</form>
`;
